web/server/codechecker_server/api/report_server.py

```diff
@@ -2660,29 +2669,22 @@ def get_analyzer_name(report):
         disabled_checkers -= all_report_checkers
         enabled_checkers |= all_report_checkers
 
-        reports_to_delete = set()
-        for bug_hash, reports in hash_map_reports.items():
-            if bug_hash in new_bug_hashes:
-                reports_to_delete.update([x.id for x in reports])
-            else:
-                for report in reports:
-                    # We set the fix date of a report only if the report
-                    # has not been fixed before.
-                    if report.fixed_at:
-                        continue
-
-                    checker = report.checker_id
-                    if checker in disabled_checkers:
-                        report.detection_status = 'off'
-                    elif checker_is_unavailable(checker):
-                        report.detection_status = 'unavailable'
-                    else:
-                        report.detection_status = 'resolved'
+        for reports in hash_map_reports.values():
+            for report in reports:
+                # We set the fix date of a report only if the report
+                # has not been fixed before.
```

Comment:
But how do we know if it was not fixed ever before? When it gets to the reopen state the fix date will be unset, do we collect somewhere else that the report was in the fixed state?

I just see in the code that the report is in currently in the fixed state and then it is skipped from the detection_status update.
Another question how can a report get into the reopened state? Is that updated somewhere else?